exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -216,9 +202,9 @@ func (r *MachinePoolReconciler) reconcileBootstrap(ctx context.Context, cluster
 		)
 
 		if !ready {
-			log.V(2).Info("Bootstrap provider is not ready, requeuing")
+			log.Info("Waiting for bootstrap provider to generate data secret and report status.ready", bootstrapConfig.GetKind(), klog.KObj(bootstrapConfig))
 			m.Status.BootstrapReady = ready
-			return ctrl.Result{RequeueAfter: externalReadyWait}, nil
```

Comment:
I wonder if there's a generic convenience we could write for all controller reconciles "do I already have a watch on this? then just return a nil response, if not, requeue"

Reply:
I think it's hard to do and I would prefer us being cautious about every requeue we add.
The problem is that having a watch or not is not enough to decide if we should just return or use requeue after.

We also have to think about if we always get an update if the condition that we are checking changes, e.g. we check here if a field in BootstrapConfig is ready, this will only every change if the BootstrapConfig is updated.

But then there are also other concerns like that this only actually works this way without race conditions because we don't cache Unstructured. If we would do this with a cached object (e.g. Cluster) it could be that we reconcile a Cluster update but the Cluster update we are retrieving from our cached client is stale.

Reply:
Understood. I suppose the maintainer needs to know that anything that pass into `reconcileExternal` here will get a watcher instantiated for free (just thinking in these lines in particular it's not super clear that there _is_ a watcher on BootstrapConfig).